gnovm/pkg/gnolang/realm.go

```diff
@@ -734,6 +766,10 @@ func (rlm *Realm) saveUnsavedObjects(store Store) {
 			continue
 		} else {
 			if !uo.GetIsDeleted() {
+				// Perform private ownership check.
+				if hasPrivateDeps(uo, rlm, store) {
+					panic("cannot persist reference of object from private realm")
+				}
```

Comment:
besides, If a dependency check is inevitable, would it be better to perform it earlier? For example, if we can determine whether an object is private before DidUpdate,  and panic in DidUpdate? cc: @thehowl for more insights.